pointcloud/basic_octree.go

```diff
@@ -182,3 +187,78 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// Pose returns the pose of the octree.
+func (octree *BasicOctree) Pose() spatialmath.Pose {
+	// TODO
+	return nil
+}
+
+// AlmostEqual compares the octree with another geometry and checks if they are equivalent.
+func (octree *BasicOctree) AlmostEqual(geom spatialmath.Geometry) bool {
+	// basically no scenario where two octrees would be equal
+	return false
+}
+
+// Transform transforms the octree by the given pose.
+func (octree *BasicOctree) Transform(p spatialmath.Pose) spatialmath.Geometry {
+	// TODO
+	return nil
+}
+
+// ToProtobuf converts the octree to a Geometry proto message.
+func (octree *BasicOctree) ToProtobuf() *commonpb.Geometry {
+	// TODO
+	return nil
```

Comment:
Originally when we put these functions on the interface, the thought was "Oh, if someone implements their own geometry type, then it'll need to know how to turn itself into protobuf" or something along those lines.

This doesn't really hold, though, because the actual proto geometry primitives aren't there.

So possibly this can also come off the interface and just be a public function, `spatialmath.ToProtobuf(geom Geometry)`

Reply:
I think for the sake of this PR lets keep them in the interface.  I'll make a ticket now to update the interface though